pyquil/paulis.py

```diff
@@ -403,11 +419,12 @@ def coalesce(d):
             for k in sorted(d):
                 term_list = d[k]
                 if (len(term_list) == 1 and not
-                   np.isclose(term_list[0].coefficient, 0.0)):
+                   (term_list[0].coefficient.real == 0 and \
```

Comment:
Please don't compare floats to `0.0`. Floating point precision will bite you otherwise.